### ISSUE TITLE
Even better performance? Eliminating the lag spikes

### ISSUE THREAD

**Issue by Vectorial1024** [Vectorial1024/EliteBionicsFramework#53]
Calculating max HP in this mod is very expensive. Currently, we cache those values and recalculate them every once in a while. This works, but it can cause lag spikes when the cache-RNG tells every value to recalculate at the same time.

Thus, another approach: what if we can know the exact factors affecting max HP and cache values on status change? EG, if max HP depends on body size, then we can just cache the values until the body size is changed. If max HP depends on the add/remove of hediffs, then we can just track the add/remove of hediffs and cache those values forever.

This requires a very good sense of correctness in dealing with this.

**Comment by PavelKer** [comment]
The performance analyzer stack trace I promised:
![ebf](https://user-images.githubusercontent.com/130398166/230969353-dfee5ff4-520c-49d1-a0b5-46aea3f01ed9.png)
It slightly varies from spike to spike, but the lines beginning with EBF are always the same. I do run lots of other mods, just to take that into account though.
Regarding combat, it does not impact me, I experience the spikes all the time, pretty much constant stuttering. I have one pawn that has like 4 bionics/implants, probably the one causing it.
The stuttering happens constantly and causes the average play time to more than double, while I have just 4 colonists currently with not too much on the map.

**Comment by PavelKer** [comment]
Just as an update, adding CONN did not eliminate the lag.

**Comment by Vectorial1024** [comment]
> Just as an update, adding CONN did not eliminate the lag.

This is actually reassuring, because it tells me that this problem has existed for a long time yet I did not notice it. To be fair, at some point, I also noticed the lag spikes, but my dumb ass did not link the lag spike with EBF...

Will need some time for this, currently focusing on something else (Real Time Clock Plus).

**Comment by PavelKer** [comment]
Rolling back to the latest GitHub version of 5.0.8.5 gets rid of the lag completely, as others mentioned too.

**Comment by Vectorial1024** [comment]
> Rolling back to the latest GitHub version of 5.0.8.5 gets rid of the lag completely

Eventually, I found a possible reason:

Remember when one of the changelogs said sth like this: "experimental; will not throw errors if mods are installed BUT not enabled"? It turns out, due to config problems, the updated DLL was actually the RW1.3 DLL, so everyone using RW1.4 DLL actually have no changed behavior.

Then, because at that time I did not know what I was doing, the updated mod detection logic was actually incorrect: it was detecting something else entirely, and this (wrong) detection is slow AF.

Now that I know what's up, I can combine this bugfix with some optimization ideas that I have in my mind and release yet another update.
